controllers/dockermachine_controller.go

```diff
@@ -123,7 +123,8 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 	if err != nil {
 		return ctrl.Result{}, err
 	}
-	if err := r.Client.Update(ctx, dockerMachine); err != nil {
+
+	if err := r.Client.Patch(ctx, dockerMachine, patch); err != nil {
```

Comment:
unfortunately there is a bug in this client's patch call that requires slightly more work than this. Please see

https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/blob/master/controllers/kubeadmconfig_controller.go#L289

Reply:
edit: What jason said. You can delete this line instead of modifying

Reply:
Ah ok 👀

Reply:
So the defer is handling the patchMachine, so the r.Client.Patch can simply be removed. That seems to follow with https://github.com/kubernetes-sigs/cluster-api-bootstrap-provider-kubeadm/pull/107.